Panic with istio 1.24-alpha.96593644e05baa5c1b6aaab3647b2d9335a23942

```
k get pod   -n kmesh-system 
NAME          READY   STATUS             RESTARTS      AGE
kmesh-5dcnv   0/1     CrashLoopBackOff   3 (24s ago)   2m28s
```

time="2024-09-05T04:20:28Z" level=warning msg="exceed the max port count,current only support maximum of 10 ports" subsys=workload_controller
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x556bffcab31f]

goroutine 101 [running]:
kmesh.net/kmesh/pkg/controller/workload.(*Processor).handleService(0xc0000c91a0, 0xc000c08210)
	/kmesh/pkg/controller/workload/workload_processor.go:506 +0xdf
kmesh.net/kmesh/pkg/controller/workload.(*Processor).handleAddressTypeResponse(0xc0000c91a0, 0xc0001124d0)
	/kmesh/pkg/controller/workload/workload_processor.go:573 +0x4d2


image 

```
      image: ghcr.io/kmesh-net/kmesh:latest
      imageID: ghcr.io/kmesh-net/kmesh@sha256:9ec2befa7f60f336a5c667ab35234a8d061af92b20ff3edd65e67be3e63613b1

```

--- a/pkg/controller/workload/workload_processor.go
+++ b/pkg/controller/workload/workload_processor.go
@@ -456,7 +456,7 @@ func (p *Processor) storeServiceData(serviceName string, waypoint *workloadapi.G
 
 	newValue := bpf.ServiceValue{}
 	newValue.LbPolicy = LbPolicyRandom
-	if waypoint != nil {
+	if waypoint != nil && waypoint.GetAddress() != nil {
 		nets.CopyIpByteFromSlice(&newValue.WaypointAddr, waypoint.GetAddress().Address)
 		newValue.WaypointPort = nets.ConvertPortToBigEndian(waypoint.GetHboneMtlsPort())
 	}
@@ -501,7 +501,7 @@ func (p *Processor) handleService(service *workloadapi.Service) error {
 	}
 
 	// Preprocess service, remove the waypoint from waypoint service, otherwise it will fall into a loop in bpf
-	if service.Waypoint != nil {
+	if service.Waypoint != nil && service.GetWaypoint().GetAddress() != nil && len(service.Addresses) != 0 {
 		// Currently istiod only set the waypoint address to the first address of the service
 		// When waypoints of different granularities are deployed together, the only waypoint service to be determined
 		// is whether it contains port 15021, ref: https://github.com/kmesh-net/kmesh/issues/691

--- a/pkg/controller/workload/workload_processor_test.go
+++ b/pkg/controller/workload/workload_processor_test.go
@@ -18,6 +18,7 @@ package workload
 
 import (
 	"net/netip"
+	"strings"
 	"testing"
 
 	service_discovery_v3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
@@ -147,6 +148,21 @@ func Test_handleWorkload(t *testing.T) {
 	checkNotExistInFrontEndMap(t, fakeSvc.Addresses[0].Address, p)
 
 	hashNameClean(p)
+}
+
+func Test_handleServiceWithWaypoint(t *testing.T) {
+	// Mainly used to test whether processor can correctly handle
+	// different types of waypoint address without panic.
+	workloadMap := bpfcache.NewFakeWorkloadMap(t)
+	p := newProcessor(workloadMap)
+
+	// Waypoint with network address.
+	svc1 := createFakeService("svc1", "10.240.10.1", "10.240.10.200")
+	// Waypoint with hostname.
+	svc2 := createFakeService("svc2", "10.240.10.1", "gateway.example.com/default")
+
+	assert.NoError(t, p.handleService(svc1))
+	assert.NoError(t, p.handleService(svc2))
 }
 
 func Test_hostnameNetworkMode(t *testing.T) {
@@ -357,6 +373,29 @@ func createFakeWorkload(ip string, networkMode workloadapi.NetworkMode) *workloa
 }
 
 func createFakeService(name, ip, waypoint string) *workloadapi.Service {
+	var w *workloadapi.GatewayAddress
+	res := strings.Split(waypoint, "/")
+	if len(res) == 2 {
+		w = &workloadapi.GatewayAddress{
+			Destination: &workloadapi.GatewayAddress_Hostname{
+				Hostname: &workloadapi.NamespacedHostname{
+					Namespace: res[1],
+					Hostname:  res[0],
+				},
+			},
+			HboneMtlsPort: 15008,
+		}
+	} else {
+		w = &workloadapi.GatewayAddress{
+			Destination: &workloadapi.GatewayAddress_Address{
+				Address: &workloadapi.NetworkAddress{
+					Address: netip.MustParseAddr(waypoint).AsSlice(),
+				},
+			},
+			HboneMtlsPort: 15008,
+		}
+	}
+
 	return &workloadapi.Service{
 		Name:      name,
 		Namespace: "default",
@@ -380,14 +419,7 @@ func createFakeService(name, ip, waypoint string) *workloadapi.Service {
 				TargetPort:  82,
 			},
 		},
-		Waypoint: &workloadapi.GatewayAddress{
-			Destination: &workloadapi.GatewayAddress_Address{
-				Address: &workloadapi.NetworkAddress{
-					Address: netip.MustParseAddr(waypoint).AsSlice(),
-				},
-			},
-			HboneMtlsPort: 15008,
-		},
+		Waypoint: w,
 	}
 }
 

